ql/src/java/org/apache/hadoop/hive/ql/parse/type/RexNodeExprFactory.java

```diff
@@ -165,22 +165,23 @@ protected RexNode createColumnRefExpr(ColumnInfo colInfo, List<RowResolver> rowR
 
   private int getPosition(ColumnInfo colInfo, List<RowResolver> rowResolverList)
       throws SemanticException {
-    ColumnInfo tmp;
-    ColumnInfo cInfoToRet = null;
     int position = 0;
-    for (RowResolver rr : rowResolverList) {
-      tmp = rr.get(colInfo.getTabAlias(), colInfo.getAlias());
-      if (tmp != null) {
-        if (cInfoToRet != null) {
-          throw new CalciteSemanticException("Could not resolve column name");
-        }
-        cInfoToRet = tmp;
-        position += rr.getPosition(cInfoToRet.getInternalName());
-      } else if (cInfoToRet == null) {
+
+    for (RowResolver rr: rowResolverList) {
+      ColumnInfo tmp = rr.get(colInfo.getTabAlias(), colInfo.getAlias());
+      if (tmp == null) {
+        // if column in not present in the RR, increment position by size of RR
```

Comment:
typo?
is not present

Reply:
Yes! I will fix it! Thanks for catching it :)